client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/ManageApplicationsPrivilege.java

```diff
@@ -0,0 +1,150 @@
+            });
+
+    static {
+        @SuppressWarnings("unchecked")
+        final ConstructingObjectParser<ManageApplicationsPrivilege, Void> apps_parser =
+            new ConstructingObjectParser<>("apps_parser", false, constructorObjects -> {
+                    final Collection<String> applications = (Collection<String>) constructorObjects[0];
+                    return new ManageApplicationsPrivilege(applications);
+                });
+        apps_parser.declareStringArray(constructorArg(), APPLICATIONS);
+
+        final ConstructingObjectParser<ManageApplicationsPrivilege, Void> scope_parser =
+                new ConstructingObjectParser<>("scope_parser", false, constructorObjects -> {
+                        return (ManageApplicationsPrivilege) constructorObjects[0];
+                    });
+        scope_parser.declareObject(constructorArg(), apps_parser, SCOPE);
+
+        PARSER.declareObject(constructorArg(), scope_parser, CATEGORY);
+    }
+
+    private final Set<String> applications;
+
+    private ManageApplicationsPrivilege(Collection<String> applications) {
+        // we do all null checks inside the constructor
+        if (null == applications || applications.isEmpty()) {
+            throw new IllegalArgumentException("managed applications list should not be null");
+        }
+        this.applications = Collections.unmodifiableSet(new HashSet<>(applications));
+    }
+
+    public Set<String> getApplications() {
+        return applications;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        ManageApplicationsPrivilege that = (ManageApplicationsPrivilege) o;
+        return applications.equals(that.applications);
+    }
+
+    @Override
+    public int hashCode() {
+        return applications.hashCode();
+    }
+
+    @Override
+    public String toString() {
+        final StringBuilder sb = new StringBuilder(getClass().getSimpleName()).append("[");
+        sb.append(APPLICATIONS.getPreferredName()).append("=[").append(Strings.collectionToCommaDelimitedString(applications));
+        sb.append("]]");
+        return sb.toString();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.startObject(CATEGORY.getPreferredName());
+        builder.startObject(SCOPE.getPreferredName());
+        builder.field(APPLICATIONS.getPreferredName(), applications);
+        builder.endObject();
+        builder.endObject();
+        return builder.endObject();
+    }
+
+    public static ManageApplicationsPrivilege fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
```

Comment:
_This_ Builder seems unnecessary.

Reply:
It is unnecessary. It is here for consistency.

Reply:
Im much happier with less code than more consistency ;)

Reply:
Noted.